packages/amplify-graphql-model-transformer/src/__tests__/model-transformer.test.ts

```diff
@@ -1906,4 +1906,239 @@ describe('ModelTransformer:', () => {
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should include autoId when timestamps are null', async () => {
+        const schema = `
+          type Post @model(timestamps: null) {
+              id: ID!
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer()],
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should include autoId when id is type String', async () => {
+        const schema = `
+          type Post @model {
+              id: String
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer()],
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should include autoId when id is type Int, Float, or Boolean', async () => {
+        const schema = `
+          type Foo @model {
+              id: Int
+              title: String!
+          }
+
+          type Bar @model {
+              id: Float
+              title: String!
+          }
+
+          type Baz @model {
+              id: Boolean
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer()],
+        });
+        expect(out.resolvers['Mutation.createFoo.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+        expect(out.resolvers['Mutation.createBar.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+        expect(out.resolvers['Mutation.createBaz.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should not include autoId when using a custom primary key', async () => {
+        const schema = `
+          type Post @model {
+              postId: ID! @primaryKey
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer(), new PrimaryKeyTransformer()],
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+    });
+
+    describe('sql', () => {
+      it('should include autoId for basic ID', async () => {
```

Comment:
Is it expected that we would generate an autoId for an explicitly declared `id` field even though there is an explicit primaryKey?

Reply:
Same question here. Also the schema used is different from the same case in DDB which has the `postId: ID! @primaryKey` defined. Does the DDB act the same as SQL with the same schema in such case?

Reply:
Just see the comments below. Looks like the DDB is the same and changing the behavior is breaking.

Reply:
The schema for SQL is different because SQL schemas require an `@primaryKey`.

```
InvalidDirectiveError: SQL model "Post" must contain a primary key field
```

Reply:
I've just checked again to make sure there are not breaking changes. I reverted my changes, but kept the tests cases. The failing tests cases are:

* autoId › dynamodb › should include autoId when timestamps are null with explicit id
  * expected: change to timestamp: null handling
* autoId › dynamodb › should include autoId when timestamps are null with implicit id
  * expected: change to timestamp: null handling
* autoId › dynamodb › should not include autoId when id is type Int, Float, or Boolean
  * expected: change to non ID or String types
* autoId › dynamodb › should not include autoId when using a custom primary key
  * expected: Removed the auto id statement because implicit id will not be added when using primary key
* autoId › sql › should include autoId when timestamps are null
  * expected: change to timestamp: null handling
* autoId › sql › should not include autoId when id is type Int, Float, or Boolean
  * expected: change to non ID or String types